Sources/TransferMethod/AddTransferMethodPresenter.swift

```diff
@@ -77,15 +77,11 @@ final class AddTransferMethodPresenter {
                         return
                     }
 
-                    guard let result = result else {
+                    guard let result = result, let fields = result.fields() else {
                         return
                     }
-                    let transferTypeDetail = result.populateTransferMethodTypeDetail(
-                        country: strongSelf.country,
-                        currency: strongSelf.currency,
-                        profileType: strongSelf.profileType,
-                        transferMethodType: strongSelf.transferMethodType)
-                    strongSelf.view.showTransferMethodFields(result.fields(), transferTypeDetail)
+                    let transferTypeDetail = TransferMethodTypeDetail(strongSelf.currency, result.transferMethodType()!)
```

Comment:
No need of TransferMethodTypeDetail, instead use result.transferMethodType(). HyperwalletTransferMethodType contains transferMethodType, fees and processingTime.

Reply:
we need `currency` in string formatting as well

Reply:
Currency is part of HyperwalletFee so you can use it from there

Reply:
fixed